### PR TITLE
fix lhm where code is pushing nulls to array of list

### PR DESCRIPTION
**Pull request by peterpark-db**
There is a bug in the LHM demo where it pushes nulls to array of list, causing error:
```
Can not merge type `ArrayType` and `DoubleType`.
```

Also made few changes:
1. Uncomment SLICING_EXPRS
2. Check if drift table exists

### CODE REVIEW

File: product_demos/DBSQL-Datawarehousing/lakehouse-monitor/01-Timeseries-monitor.py [hunk 1]
```diff
@@ -121,6 +121,10 @@
 
 # Display the drift metrics table
 drift_table = f"{TABLE_NAME}_drift_metrics"
+
+while not spark.catalog.tableExists(drift_table) or spark.table(drift_table).isEmpty():
```
Comment (QuentinAmbard) [inline product_demos/DBSQL-Datawarehousing/lakehouse-monitor/01-Timeseries-monitor.py:125]: can you do it in a range to make sure it doesn't loop forever?

**Comment by QuentinAmbard** [comment]
@aelhelouDB could you help reviewing this PR ?

**Review by aelhelouDB — CHANGES_REQUESTED** [review]
- Use monitor API to check if monitor/refresh has finished (instead of verifying if drift table exists... because it can exist anyway but without being refreshed)
- Accept proposed work-around to mitigate type-merge issue but keep a TODO to make it more robust/consistent later

File: product_demos/DBSQL-Datawarehousing/lakehouse-monitor/01-Timeseries-monitor.py (hunk 1 above)
Comment (peterpark-db) [reply product_demos/DBSQL-Datawarehousing/lakehouse-monitor/01-Timeseries-monitor.py]: Committed the suggestion but also removed the check for the profile table that was there before

File: product_demos/DBSQL-Datawarehousing/lakehouse-monitor/_resources/01-DataGeneration.py [hunk 2]
```diff
@@ -554,7 +549,10 @@ def inject_issues(df, campaign_start_dates):
     steady_null_columns = ['ProductTags', 'ShippingAddress', 'Wishlist', 'GiftWrap']
     for column in steady_null_columns:
         null_indices = df.sample(frac=random.uniform(0.1, 0.15)).index
-        df.loc[null_indices, column] = np.nan
+        if column in ['ProductTags', 'Wishlist']:
```
[the remark this answers is not in the document]
Comment (peterpark-db) [reply product_demos/DBSQL-Datawarehousing/lakehouse-monitor/_resources/01-DataGeneration.py:552]: Yes, this is the fix for the type merge

**Pull request merged**